playwright-action/helpers/playwrightSetup.js.js

```diff
@@ -0,0 +1,57 @@
+//@ts-check
+const { chromium } = require("@playwright/test");
+const fs = require('fs');
+const path = require('path');
+
+//utility func to generate css selector and return id and strings
+function waitFor(str) {
+      return `@${str}`;
+}
+
+function id(str) {
+      return `#${str}`;
+}
+
+
+
+//create a fresh testing env for testcases
+async function InitialEnvSetup() {
+      const browser = await chromium.launch();
+      const context = await browser.newContext();
+      const page = await context.newPage();
+
+      return { browser, context, page };
+}
+
+
+//capture and save a screenshot
+async function captureAndSaveScreenshot(page, filename) {
+      await page.screenshot({ path: filename });
+}
+
+//Save meshmapscreenshot and generate a download link.    
+async function saveGraph(customUrl) {
+      const browser = await chromium.launch();
+      const page = await browser.newPage();
+    
+      await page.goto(customUrl);
+    
+      // Capture a screenshot and save it with a custom filename
+      const date = new Date();
+      // Format: MeshMap-Fri Sep 09 2023-10:30:45 AM.png
+      const fileName = `MeshMap-${date.toDateString()}-${date.toLocaleTimeString()}.png`;
```

Comment:
`toDateString()` will give a string with some spaces in it, r8?
The same goes for `toLocaleTimeString`.

Can we format it to YYYYMMDDHHMMSS or related formats which is less ambiguous? Is it possible?

Reply:
There should be a way. My js isnt that solid but i'll fix it

Reply:
```JS
const getFormattedDateTime = () => {
  const now = new Date();

  // Extract date components (YYYY_MM_DD format)
  const year = now.getFullYear();

  // Add 1 because months are 0-indexed
  const month = (now.getMonth() + 1).toString().padStart(2, "0");
  const day = now.getDate().toString().padStart(2, "0");

  // Extract time components (HH_MM_SS format)
  const hours = now.getHours().toString().padStart(2, "0");
  const minutes = now.getMinutes().toString().padStart(2, "0");
  const seconds = now.getSeconds().toString().padStart(2, "0");

  // Concatenate date and time components into the desired format
  const formattedString = `Meshmap_${year}_${month}_${day}_${hours}_${minutes}_${seconds}`;

  return formattedString;
};
```

Reply:
![image](https://github.com/layer5labs/meshmap-snapshot/assets/34866732/86ef2d68-4c20-4560-89cd-1271d2f95d8a)